can-route-hash.js

```diff
@@ -11,9 +11,10 @@ var SimpleObservable = require("can-simple-observable");
 
 var domEvents = require("can-dom-events");
 
-function getHash(){
+function getHash(root){
     var loc = LOCATION();
-    return loc.href.split(/#!?/)[1] || "";
+    const re = new RegExp(root +"?");
```

Comment:
CanJS is written in ES5, so this needs to be `var`.

This also changes root to work as a part of a regular expression. 

The behavior of `root` isn't documented: https://canjs.com/doc/can-route-hash.html so we can do whatever we want here.  But we should think if using it as part of a regular expression is the right thing, or should `root` __BE__ a regular expression itself.


We'd have to see where else `root` is used.  Is it used as a string everywhere else?

Reply:
So - yes. Its used as a string throughout `can-route-hash`. Not sure if its used elsewhere. I don't think so.

I think the re can be removed and replaced with root. I'll push that.